atlassian_jwt_auth/contrib/aiohttp/key.py

```diff
@@ -21,8 +22,16 @@ def _get_session(self):
         return aiohttp.ClientSession(loop=self.loop)
 
     async def _retrieve(self, url, requests_kwargs):
-        resp = await self._session.get(url, headers={'accept': PEM_FILE_TYPE},
-                                       **requests_kwargs)
-        resp.raise_for_status()
-        self._check_content_type(url, resp.headers['content-type'])
-        return await resp.text()
+        try:
+            resp = await self._session.get(url, headers={'accept':
+                                                         PEM_FILE_TYPE},
+                                           **requests_kwargs)
+        except aiohttp.ClientError as e:
+            raise requests.RequestException(e)
+
+        try:
+            resp.raise_for_status()
+            self._check_content_type(url, resp.headers['content-type'])
+            return await resp.text()
+        except aiohttp.http.HttpProcessingError as e:
+            raise requests.HTTPError(e)
```

Comment:
Why wrap this exception?

Reply:
None of the code that catches key retrieval exceptions or request exceptions handles aiohttp exceptions. Normalizing to one type of exception fixes those bugs and makes for easier code reuse.

I'm not set on `requests.HTTPError` here: we could use something else or create a new exception class.

Reply:
Good catch!

Reply:
I am not going to agree on this change because you will make it harder for users of the `aiohttp` code to obtain or catch a specific exception that might occur. 

Importantly, 
* `aiohttp.ClientError` extends `Exception` and does not have a `request` nor a `response` attribute.
* `aiohttp.HttpProcessingError` extends `Exception` and has different attributes than `requests.HTTPError`.

Reply:
> If different callers need to have different exceptions, then they probably need to have a custom interface as well.

Right so just to clarify I wasn't thinking that a `django` or` flask` application would use the <django/flask> contrib code with an `asyncio` `HTTPSPublicKeyRetriever`. Perhaps you are thinking of this?

Reply:
I was thinking that `HTTPSPublicKeyRetriever` shouldn't have to care what's calling it, so long as both sides follow the same interface.